e2e/webhook_test.go

```diff
@@ -123,6 +125,56 @@ func TestSecretValueInjection(t *testing.T) {
 	testenv.Test(t, secret, configMap)
 }
 
+func TestDockerAuthWithJsonKey(t *testing.T) {
```

Comment:
We have two methods in this test-file:

- One for testing secret/config map mutation.
- One for testing pod mutation.

This test should be part of the first one, allowing you to reuse that logic there.